BlueprintUI/Sources/Layout/Builder/ElementBuilder.swift

```diff
@@ -0,0 +1,29 @@
+/// Generic result builder for converting blocks of `Child...` into `[Child]`.
+@_functionBuilder // change to @resultBuilder when we upgrade to Swift 5.4.
+public struct ElementBuilder {
```

Comment:
Thinking out loud: It would be nice if we could avoid triple-duplicating these; since they're all nearly the same. Could we introduce generics here around the content type, and defer to that content type for the bottom two methods:

```
    public static func buildExpression(_ element: Element) -> Component {
        [element.gridRowChild(width: .proportional(1))]
    }

    public static func buildExpression(_ elements: [Element]) -> Component {
        elements.map { $0.gridRowChild(width: .proportional(1)) }
    }
```

Which seem to be the two implementations which diverge across the builder types?

(Or maybe this is basically just boilerplate that will never change; and having three is fine! Thoughts?)

Reply:
Love this idea...adding a generic builder definitely cleans up the code. I created a protocol to factor out the differing logic.